src/sentry/tasks/assemble.py

```diff
@@ -311,14 +326,91 @@ def _create_artifact_bundle(
                     artifact_bundle=artifact_bundle,
                     source_file_type=source_file_type.value,
                 )
+        else:
+            raise AssembleArtifactsError(
+                "uploading a bundle without debug ids or release is prohibited"
+            )
+
+
+def get_with_manifest_fallback(value, manifest_key, manifest, throw=True):
+    if value is None:
+        manifest_value = manifest.get(manifest_key)
+        if manifest_value is None and throw:
+            raise AssembleArtifactsError(f"failed to load {manifest_key} from the manifest")
+
+        return manifest_value
+
+    return value
+
+
+def handle_assemble_for_release_file(bundle, archive, manifest, organization, version):
+    org_slug = manifest.get("org")
+    if organization.slug != org_slug:
+        raise AssembleArtifactsError("organization does not match uploaded bundle")
+
+    release_name = manifest.get("release")
+    if release_name != version:
+        raise AssembleArtifactsError("release does not match uploaded bundle")
+
+    try:
+        release = Release.objects.get(organization_id=organization.id, version=release_name)
+    except Release.DoesNotExist:
+        raise AssembleArtifactsError("release does not exist")
 
-            return True
+    dist_name = manifest.get("dist")
+    dist = None
+    if release and dist_name:
+        dist = release.add_dist(dist_name)
 
-    return False
+    artifact_count = len(manifest.get("files", {}))
+    min_artifact_count = options.get("processing.release-archive-min-files")
+    saved_as_archive = False
+
+    if artifact_count >= min_artifact_count:
+        try:
+            update_artifact_index(release, dist, bundle)
+            saved_as_archive = True
+        except Exception as exc:
+            logger.error("Unable to update artifact index", exc_info=exc)
+
+    if not saved_as_archive:
+        meta = {
+            "organization_id": organization.id,
+            "release_id": release.id,
+            "dist_id": dist.id if dist else dist,
+        }
+        _store_single_files(archive, meta, True)
+
+    return artifact_count
+
+
+def handle_assemble_for_artifact_bundle(bundle, manifest, organization, version, project_ids, dist):
+    # We want to give precedence to the request fields and only if they are unset fallback to the manifest's
+    # contents.
+    release_name = get_with_manifest_fallback(version, "release", manifest, throw=False)
+    dist_name = get_with_manifest_fallback(dist, "dist", manifest, throw=False)
```

Comment:
Why so overengineered? 😅

```suggestion
    release_name = version or manifest.get("release")
    dist_name = dist or manifest.get("dist")
```

Reply:
I have actually used the `throw` in the `handle_assemble_for_release_file` method but then decided to keep the old logic since the previous implementation required consistent values across request/manifest instead of having the manifest as a fallback and I forgot to remove this convoluted function. Thanks!